### PR TITLE
Add more architectures.

### PR DESCRIPTION
**Pull request by hameerabbasi**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+build_platform: {osx_arm64: osx_64, linux_ppc64le: linux_64, linux_aarch64: linux_64}
```
Comment (isuruf) [inline recipe/conda_build_config.yaml:1]: This should be in `conda-forge.yml` file at the root of the repo

Comment (hameerabbasi) [reply recipe/conda_build_config.yaml]: Done in c2c051059be4defb2908cd846c8270a2299ac4b1.

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

**Comment by hameerabbasi** [comment]
@conda-forge-admin, rerender

**Comment by hameerabbasi** [comment]
@isuruf I don't understand the error message, `Python3_NumPy_INCLUDE_DIRS` is passed correctly to CMake.

```
2024-07-22T05:31:28.3268554Z + cmake -DCMAKE_AR=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_build_env/bin/aarch64-conda-linux-gnu-ar -DCMAKE_CXX_COMPILER_AR=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_build_env/bin/aarch64-conda-linux-gnu-gcc-ar -DCMAKE_C_COMPILER_AR=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_build_env/bin/aarch64-conda-linux-gnu-gcc-ar -DCMAKE_RANLIB=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_build_env/bin/aarch64-conda-linux-gnu-ranlib -DCMAKE_CXX_COMPILER_RANLIB=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_build_env/bin/aarch64-conda-linux-gnu-gcc-ranlib -DCMAKE_C_COMPILER_RANLIB=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_build_env/bin/aarch64-conda-linux-gnu-gcc-ranlib -DCMAKE_LINKER=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_build_env/bin/aarch64-conda-linux-gnu-ld -DCMAKE_STRIP=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_build_env/bin/aarch64-conda-linux-gnu-strip -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_ROOT_PATH_MODE_PROGRAM=NEVER -DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=ONLY -DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=ONLY '-DCMAKE_FIND_ROOT_PATH=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_;/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_build_env/aarch64-conda-linux-gnu/sysroot' -DCMAKE_INSTALL_PREFIX=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_ -DCMAKE_INSTALL_LIBDIR=lib '-DCMAKE_PROGRAM_PATH=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_build_env/bin;/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/bin' -DCMAKE_SYSTEM_NAME=Linux -DCMAKE_SYSTEM_PROCESSOR=aarch64 -DLLVM_TABLEGEN_EXE=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_build_env/bin/llvm-tblgen -DNATIVE_LLVM_DIR=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_build_env/lib/cmake/llvm '-DCROSS_TOOLCHAIN_FLAGS_NATIVE=-DCMAKE_C_COMPILER=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_build_env/bin/x86_64-conda-linux-gnu-cc;-DCMAKE_CXX_COMPILER=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_build_env/bin/x86_64-conda-linux-gnu-c++;-DCMAKE_C_FLAGS=-O2;-DCMAKE_CXX_FLAGS=-O2;-DCMAKE_EXE_LINKER_FLAGS="-L/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_build_env/lib";-DCMAKE_MODULE_LINKER_FLAGS=;-DCMAKE_SHARED_LINKER_FLAGS=;-DCMAKE_STATIC_LINKER_FLAGS=;-DCMAKE_AR=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_build_env/bin/aarch64-conda-linux-gnu-ar;-DCMAKE_RANLIB=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_build_env/bin/aarch64-conda-linux-gnu-ranlib;-DZLIB_ROOT=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_build_env;-DPython3_NumPy_INCLUDE_DIRS=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_build_env/venv/lib/python3.8/site-packages/numpy/core/include' -DCMAKE_INSTALL_PREFIX=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_ -DCMAKE_BUILD_TYPE=Release -DLLVM_BUILD_LLVM_DYLIB=ON -DLLVM_LINK_LLVM_DYLIB=ON -DLLVM_BUILD_TOOLS=ON -DLLVM_BUILD_UTILS=ON -DMLIR_ENABLE_BINDINGS_PYTHON=ON -DPython3_EXECUTABLE=/home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1721625845367/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/bin/python -GNinja ../mlir
2
```

File: recipe/build.sh [hunk 2]
```diff
@@ -10,14 +10,15 @@ fi
 if [[ "${CONDA_BUILD_CROSS_COMPILATION:-0}" == "1" ]]; then
   CMAKE_ARGS="${CMAKE_ARGS} -DLLVM_TABLEGEN_EXE=$BUILD_PREFIX/bin/llvm-tblgen -DNATIVE_LLVM_DIR=$BUILD_PREFIX/lib/cmake/llvm"
   CMAKE_ARGS="${CMAKE_ARGS} -DCROSS_TOOLCHAIN_FLAGS_NATIVE=-DCMAKE_C_COMPILER=$CC_FOR_BUILD;-DCMAKE_CXX_COMPILER=$CXX_FOR_BUILD;-DCMAKE_C_FLAGS=-O2;-DCMAKE_CXX_FLAGS=-O2;-DCMAKE_EXE_LINKER_FLAGS=\"-L$BUILD_PREFIX/lib\";-DCMAKE_MODULE_LINKER_FLAGS=;-DCMAKE_SHARED_LINKER_FLAGS=;-DCMAKE_STATIC_LINKER_FLAGS=;-DCMAKE_AR=$(which ${AR});-DCMAKE_RANLIB=$(which ${RANLIB});-DZLIB_ROOT=${BUILD_PREFIX}"
+  CMAKE_ARGS="${CMAKE_ARGS} -DPython3_NumPy_INCLUDE_DIRS=$($PYTHON -c 'import numpy as np; print(np.get_include())')"
```
Comment (hameerabbasi) [inline recipe/build.sh:13]: Will this work correctly when cross-compiling? In other words, will the content and location of the include directory be correct?

**Review by hameerabbasi — COMMENTED** [review]
One question

File: recipe/build.sh (hunk 2 above)
Comment (jaimergp) [reply recipe/build.sh:13]: In principle the `python_cross` stuff should handle these, I think?

**Comment by hameerabbasi** [comment]
@conda-forge-admin, rerender

**Comment by h-vetinari** [comment]
Don't mind me, I wanted to look at the CI results (which aren't there, that's why I didn't see them), and thought I was pressing on a fold-out button.

**Comment by hameerabbasi** [comment]
> Don't mind me, I wanted to look at the CI results (which aren't there, that's why I didn't see them), and thought I was pressing on a fold-out button.

Here's the current failure I've been striúggling with forever:

```
-- Found Python3: $BUILD_PREFIX/bin/python3.9 (found version "3.9.19") found components: Interpreter Development Development.Module Development.Embed
CMake Error at /home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1722244606773/_build_env/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find Python3 (missing: Python3_NumPy_INCLUDE_DIRS NumPy) (found
  version "3.9.19")
Call Stack (most recent call first):
  /home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1722244606773/_build_env/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
  /home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1722244606773/_build_env/share/cmake-3.30/Modules/FindPython/Support.cmake:3984 (find_package_handle_standard_args)
  /home/conda/feedstock_root/build_artifacts/mlir-python-bindings_1722244606773/_build_env/share/cmake-3.30/Modules/FindPython3.cmake:596 (include)
  cmake/modules/MLIRDetectPythonEnv.cmake:21 (find_package)
  CMakeLists.txt:173 (mlir_configure_python_dev_packages)
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -42,9 +43,9 @@ requirements:
     - mlir {{ version }}
 
 test:
-  imports:
-    - mlir
-    - mlir.ir
+  imports:                  # [build_platform == target_platform]
+    - mlir                  # [build_platform == target_platform]
+    - mlir.ir               # [build_platform == target_platform]
```
Comment (h-vetinari) [inline recipe/meta.yaml:48]: Leave this out the selectors here, but add
```
test: native_and_emulated
```
to `conda-forge.yml` in the feedstock root.

File: recipe/build.sh [hunk 4]
```diff
@@ -17,15 +25,32 @@ fi
...
-  -DPython3_EXECUTABLE="$PYTHON" \
+  -DMLIR_ENABLE_BINDINGS_PYTHON:bool=ON \
+  -DCMAKE_SKIP_RPATH:bool=ON \
+  -DPYTHON_PACKAGES_PATH="${SP_DIR}" \
+  -DPYTHON_EXECUTABLE="${PYTHON}" \
+  -DPYTHON_INCLUDE_DIR="${INC_PYTHON}" \
+  -DPYTHON_LIBRARY="${LIB_PYTHON}" \
+  -DPYTHON3_EXECUTABLE="${PYTHON}" \
+  -DPYTHON3_INCLUDE_DIR="${INC_PYTHON}" \
+  -DPYTHON3_NUMPY_INCLUDE_DIRS="$(python -c 'import numpy;print(numpy.get_include())')" \
```
Comment (h-vetinari) [inline recipe/build.sh:43]: This ends up picking up the numpy from the build environment, and thus the wrong architecture. Copying from the logs (and reinserting the env vars):
```
-DPYTHON3_NUMPY_INCLUDE_DIRS=$BUILD_PREFIX/venv/lib/python3.8/site-packages/numpy/core/include
                              ^^^^^^
                              !!!!!!
```
It looks like a lot of this is copied from https://github.com/conda-forge/opencv-feedstock/blob/main/recipe/build.sh, but that's not a good reference in this case. Normally you shouldn't have to override all that, as cross-python is supposed to take care of it.

File: recipe/build.sh [hunk 5]
```diff
@@ -7,6 +7,14 @@ if [[ "${target_platform}" == "linux-ppc64le" ]]; then
   export CXXFLAGS="${CXXFLAGS//-fno-plt/}"
 fi
 
+IS_PYPY=$(${PYTHON} -c "import platform; print(int(platform.python_implementation() == 'PyPy'))")
+LIB_PYTHON="${PREFIX}/lib/libpython${PY_VER}${SHLIB_EXT}"
+if [[ ${IS_PYPY} == "1" ]]; then
+    INC_PYTHON="$PREFIX/include/pypy${PY_VER}"
+else
+    INC_PYTHON="$PREFIX/include/python${PY_VER}"
+fi
```
Comment (h-vetinari) [inline recipe/build.sh:16]: Please drop the pypy bits, we're not going to do that here.

File: conda-forge.yml [hunk 6]
```diff
@@ -1,3 +1,4 @@
+build_platform: {osx_arm64: osx_64, linux_ppc64le: linux_64, linux_aarch64: linux_64}
```
Comment (h-vetinari) [inline conda-forge.yml:1]: Please use
```
build_platform:
  linux_aarch64: linux_64
  linux_ppc64le: linux_64
  osx_arm64: osx_64
```
and sort it after `bot:`

**Review by h-vetinari — COMMENTED** [review]
First look.

File: recipe/meta.yaml (hunk 3 above)
Comment (hameerabbasi) [reply recipe/meta.yaml]: Done in https://github.com/conda-forge/mlir-python-bindings-feedstock/pull/32/commits/0f43913ef8c470741cfb52d776a4be7485986430.

File: recipe/build.sh (hunk 4 above)
Comment (hameerabbasi) [reply recipe/build.sh]: Done in https://github.com/conda-forge/mlir-python-bindings-feedstock/pull/32/commits/0f43913ef8c470741cfb52d776a4be7485986430.

File: recipe/build.sh (hunk 5 above)
Comment (hameerabbasi) [reply recipe/build.sh]: Done in https://github.com/conda-forge/mlir-python-bindings-feedstock/pull/32/commits/0f43913ef8c470741cfb52d776a4be7485986430.

File: conda-forge.yml (hunk 6 above)
Comment (hameerabbasi) [reply conda-forge.yml]: Done in https://github.com/conda-forge/mlir-python-bindings-feedstock/pull/32/commits/0f43913ef8c470741cfb52d776a4be7485986430.

File: recipe/build.sh (hunk 4 above)
Comment (h-vetinari) [reply recipe/build.sh]: I meant that we should be dropping way more of the `-DPYTHON*` options here.

BTW, while we iterate, you could add `skip: true  # [build_platform == target_platform]` to the global `build:` section in `meta.yaml`; we don't need to keep dragging along the native builds

File: recipe/build.sh [hunk 7]
```diff
@@ -17,15 +20,14 @@ fi
...
+cmake ${CMAKE_ARGS} \
   -DCMAKE_INSTALL_PREFIX="${PREFIX}" \
   -DCMAKE_BUILD_TYPE=Release \
   -DLLVM_BUILD_LLVM_DYLIB=ON \
   -DLLVM_LINK_LLVM_DYLIB=ON \
   -DLLVM_BUILD_TOOLS=ON \
   -DLLVM_BUILD_UTILS=ON \
-  -DMLIR_ENABLE_BINDINGS_PYTHON=ON \
-  -DPython3_EXECUTABLE="$PYTHON" \
+  -DMLIR_ENABLE_BINDINGS_PYTHON:bool=ON \
```
Comment (h-vetinari) [inline recipe/build.sh:30]: One we need to keep is this one
```suggestion
  -DMLIR_ENABLE_BINDINGS_PYTHON:bool=ON \
  -DPython3_EXECUTABLE=$PREFIX \
```

File: recipe/build.sh (hunk 4 above)
Comment (hameerabbasi) [reply recipe/build.sh]: > while we iterate, you could add `skip: true # [build_platform == target_platform]` to the global `build:` section in `meta.yaml`;

Nice idea! However that seems to skip all builds for some reason now.

File: recipe/build.sh (hunk 7 above)
Comment (hameerabbasi) [reply recipe/build.sh:30]: Done in https://github.com/conda-forge/mlir-python-bindings-feedstock/pull/32/commits/dc187321817b95d3c5bb2cec03c93b1f413f140f.

Comment (h-vetinari) [reply recipe/build.sh:30]: Sorry, I was typing only half-awake, it should have stayed "$PYTHON", not $PREFIX.

File: recipe/build.sh [hunk 8]
```diff
@@ -17,15 +20,15 @@ fi
...
+  -DPython3_EXECUTABLE="${PYTHON}" \
```
Comment (h-vetinari) [inline recipe/build.sh:31]: Looking at the logs
```
  $BUILD_PREFIX/share/cmake-3.30/Modules/FindPython3.cmake:596 (include)
  cmake/modules/MLIRDetectPythonEnv.cmake:21 (find_package)
  CMakeLists.txt:173 (mlir_configure_python_dev_packages)
```
the upstream [code](https://github.com/llvm/llvm-project/blob/llvmorg-18.1.8/mlir/cmake/modules/MLIRDetectPythonEnv.cmake#L21) in question, and the [CMake docs](https://cmake.org/cmake/help/latest/module/FindPython3.html#hints), the first thing we should do is to set `Python3_ROOT_DIR`. This is borne out by a simple github [search](https://github.com/search?q=org%3Aconda-forge%20Python3_ROOT_DIR&type=code) in conda-forge as well.
```suggestion
  -DPython3_EXECUTABLE="${PYTHON}" \
  -DPython3_ROOT_DIR="${PREFIX}" \
```

File: recipe/build.sh [hunk 9]
```diff
@@ -17,15 +20,16 @@ fi
...
+  -DPython3_ROOT_DIR="${PREFIX}" \
```
Comment (h-vetinari) [inline recipe/build.sh:32]:
```suggestion
  -DPython3_ROOT_DIR="${PREFIX}" \
  -DPython3_FIND_STRATEGY=LOCATION \
  -DPython3_FIND_UNVERSIONED_NAMES=FIRST \
```

**Comment by h-vetinari** [comment]
It seems the pinocchio recipe has found a [work-around](https://github.com/conda-forge/pinocchio-feedstock/blob/main/recipe/build.sh) for the "not finding numpy in cross-compilation" issue.

**Comment by hameerabbasi** [comment]
> It seems the pinocchio recipe has found a [work-around](https://github.com/conda-forge/pinocchio-feedstock/blob/main/recipe/build.sh) for the "not finding numpy in cross-compilation" issue.

Okay, now we're past config and into build errors.

**Comment by h-vetinari** [comment]
The issue is that it needs a native `mlir-tblgen` that needs to run on the build_platform:
```
[94/2003] Building native mlir-tblgen...
```
which then fails because it points to `$PREFIX` instead of `$BUILD_PREFIX`. The llvmdev [recipe](https://github.com/conda-forge/llvmdev-feedstock/blob/main/recipe/build.sh) has some inspiration for this.

File: recipe/build.sh [hunk 10]
```diff
@@ -7,6 +7,18 @@ if [[ "${target_platform}" == "linux-ppc64le" ]]; then
...
+
+if [[ $CONDA_BUILD_CROSS_COMPILATION == 1 ]]; then
+  echo "Copying files from $BUILD_NUMPY_INCLUDE_DIRS to $TARGET_NUMPY_INCLUDE_DIRS"
+  mkdir -p $TARGET_NUMPY_INCLUDE_DIRS
+  cp -r $BUILD_NUMPY_INCLUDE_DIRS/numpy $TARGET_NUMPY_INCLUDE_DIRS
+fi
+
 if [[ "${CONDA_BUILD_CROSS_COMPILATION:-0}" == "1" ]]; then
   CMAKE_ARGS="${CMAKE_ARGS} -DLLVM_TABLEGEN_EXE=$BUILD_PREFIX/bin/llvm-tblgen -DNATIVE_LLVM_DIR=$BUILD_PREFIX/lib/cmake/llvm"
   CMAKE_ARGS="${CMAKE_ARGS} -DCROSS_TOOLCHAIN_FLAGS_NATIVE=-DCMAKE_C_COMPILER=$CC_FOR_BUILD;-DCMAKE_CXX_COMPILER=$CXX_FOR_BUILD;-DCMAKE_C_FLAGS=-O2;-DCMAKE_CXX_FLAGS=-O2;-DCMAKE_EXE_LINKER_FLAGS=\"-L$BUILD_PREFIX/lib\";-DCMAKE_MODULE_LINKER_FLAGS=;-DCMAKE_SHARED_LINKER_FLAGS=;-DCMAKE_STATIC_LINKER_FLAGS=;-DCMAKE_AR=$(which ${AR});-DCMAKE_RANLIB=$(which ${RANLIB});-DZLIB_ROOT=${BUILD_PREFIX}"
```
Comment (h-vetinari) [inline recipe/build.sh:24]:
```suggestion
  NATIVE_FLAGS="-DCMAKE_C_COMPILER=$CC_FOR_BUILD;-DCMAKE_CXX_COMPILER=$CXX_FOR_BUILD;-DCMAKE_C_FLAGS=-O2;-DCMAKE_CXX_FLAGS=-O2"
  NATIVE_FLAGS="${NATIVE_FLAGS};-DCMAKE_EXE_LINKER_FLAGS=-Wl,-rpath,${BUILD_PREFIX}/lib;-DCMAKE_MODULE_LINKER_FLAGS="
  NATIVE_FLAGS="${NATIVE_FLAGS};-DCMAKE_SHARED_LINKER_FLAGS=;-DCMAKE_STATIC_LINKER_FLAGS="
  NATIVE_FLAGS="${NATIVE_FLAGS};-DLLVM_ENABLE_ZSTD=OFF;-DLLVM_ENABLE_LIBXML2=OFF;-DLLVM_ENABLE_ZLIB=OFF;"
  CMAKE_ARGS="${CMAKE_ARGS} -DCROSS_TOOLCHAIN_FLAGS_NATIVE=${NATIVE_FLAGS}"
```

File: recipe/meta.yaml [hunk 11]
```diff
@@ -25,6 +27,7 @@ requirements:
     - {{ compiler('cxx') }}
     - {{ stdlib('c') }}
     - llvmdev {{ version }}  # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}  # [build_platform != target_platform]
     - python                 # [build_platform != target_platform]
     - numpy                  # [build_platform != target_platform]
     - pybind11 >=2.8.0       # [build_platform != target_platform]
```
Comment (h-vetinari) [inline recipe/meta.yaml:33]:
```suggestion
    - pybind11 >=2.8.0       # [build_platform != target_platform]
    # native mlir-tblgen wants to find these for linking
    - zlib                   # [build_platform != target_platform]
    - zstd                   # [build_platform != target_platform]
```

**Comment by h-vetinari** [comment]
OK, this is upstream (or CMake) being obtuse. It ignores the `LLVM_ENABLE_*` flags for the cross-compilation, and just uses what it found earlier during config
```
-- Check for working CXX compiler: $BUILD_PREFIX/bin/aarch64-conda-linux-gnu-c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found ZLIB: $PREFIX/lib/libz.so (found version "1.3.1")
-- Found zstd: $PREFIX/lib/libzstd.so
-- Found LibXml2: $PREFIX/lib/libxml2.so (found version "2.12.7")
```
directly on the linker path
```
[...] $BUILD_PREFIX/lib/libLLVMDemangle.a  -lrt  -ldl  -lpthread  -lm  $PREFIX/lib/libz.so  $PREFIX/lib/libzstd.so
```
I checked the last build here and the bindings here don't even link to zlib/zstd, so let's just switch that off completely.

**Comment by h-vetinari** [comment]
There were a bunch of issues; in summary:
* build script was pointing to `LLVM_TABLEGEN_EXE`, which is irrelevant here
* `LLVM_NATIVE_DIR` in contrast is essential (used by our [patch](https://github.com/conda-forge/mlir-python-bindings-feedstock/blob/main/recipe/0001-Support-cross-compiling-standalone-MLIR.patch))
* The key to pointing the native compilation to the right prefix was `-DCMAKE_PREFIX_PATH` (taken from mlir recipe where this is working already)
* In principle, we shouldn't even need to recompile `mlir-tblgen`, because we can just depend on `mlir` itself here (in obvious contrast to the `mlir` recipe)
* If we do that though, the build still tries to compile mlir-tblgen (probably related to [this](https://github.com/llvm/llvm-project/blob/llvmorg-18.1.8/llvm/cmake/modules/TableGen.cmake#L206)), so we still need the other fixes as well

However, figuring all this out has created a mighty mess in the commit history, so I'll open a rebased PR (with proper authorship of the commits of course) that cleans this up.

**Review by h-vetinari — CHANGES_REQUESTED** [review]
This is green now. :)

I have a few further necessary fixes (in addition to the clean-up) in #33 so I suggest to move the discussion there.

**Comment by hameerabbasi** [comment]
Superceded by #33.

**Pull request closed without merging**